src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -127,22 +127,38 @@ public void setResultString(String resultString) {
      * @param maxScore   maximum score reachable at corresponding exercise
      */
     public void setResultString(Double totalScore, @Nullable Double maxScore) {
-        resultString = createResultString(totalScore, maxScore);
+        resultString = createResultString(totalScore, maxScore, null);
+    }
+
+    /**
+     * Sets the resultString attribute
+     *
+     * @param totalScore total amount of scored points between 0 and maxScore
+     * @param maxScore   maximum score reachable at corresponding exercise
+     * @param exercise   corresponding exercise
+     */
+    public void setResultString(Double totalScore, @Nullable Double maxScore, Exercise exercise) {
+        resultString = createResultString(totalScore, maxScore, exercise);
     }
 
     /**
      * Builds the resultString attribute
      *
      * @param totalScore total amount of scored points
      * @param maxScore   maximum score reachable at corresponding exercise
+     * @param exercise   corresponding exercise
      * @return String with result string in this format "2 of 13 points" or "2 points"
      */
-    public String createResultString(Double totalScore, @Nullable Double maxScore) {
+    public String createResultString(Double totalScore, @Nullable Double maxScore, Exercise exercise) {
         DecimalFormat formatter = new DecimalFormat("#.##");
         if (maxScore == null) {
             return formatter.format(totalScore) + " points";
         }
         else {
+            if (exercise != null && exercise.isZeroPointExercise()) {
+                totalScore = 0.0;
+                maxScore = 0.0;
+            }
             return formatter.format(totalScore) + " of " + formatter.format(maxScore) + " points";
         }
```

Comment:
```suggestion
        if (exercise != null && exercise.isZeroPointExercise()) {
            totalScore = 0.0;
            maxScore = 0.0;
        }
        return formatter.format(totalScore) + " of " + formatter.format(maxScore) + " points";
```

You probably don't need that `else`

Reply:
👍